volume/drivers/nfs/nfs.go

```diff
@@ -356,30 +476,54 @@ func (d *driver) Mount(volumeID string, mountpath string, options map[string]str
 		return err
 	}
 
-	srcPath := path.Join(":", nfsPath, volumeID)
-	mountExists, err := d.mounter.Exists(srcPath, mountpath)
-	if !mountExists {
-		d.mounter.Unmount(path.Join(nfsPath, volumeID), mountpath,
-			syscall.MNT_DETACH, 0, nil)
-		if err := d.mounter.Mount(
-			0, path.Join(nfsPath, volumeID),
-			mountpath,
-			string(v.Spec.Format),
-			syscall.MS_BIND,
-			"",
-			0,
-			nil,
-		); err != nil {
-			logrus.Printf("Cannot mount %s at %s because %+v",
-				path.Join(nfsPath, volumeID), mountpath, err)
-			return err
+	if v.GetSpec().GetShared() {
+		if v.GetState() != api.VolumeState_VOLUME_STATE_AVAILABLE {
+			return fmt.Errorf("Volume is not in an available state")
+		}
+		// File access
+		srcPath := path.Join(":", nfsPath, volumeID)
+		mountExists, _ := d.mounter.Exists(srcPath, mountpath)
+		if !mountExists {
+			/*
+				THIS was here and probably needs to be removed
+				d.mounter.Unmount(path.Join(nfsPath, volumeID), mountpath,
+					syscall.MNT_DETACH, 0, nil)
+			*/
+			if err := d.mounter.Mount(
+				0, path.Join(nfsPath, volumeID),
+				mountpath,
+				string(v.Spec.Format),
+				syscall.MS_BIND,
+				"",
+				0,
+				nil,
+			); err != nil {
+				logrus.Printf("Cannot mount %s at %s because %+v",
+					path.Join(nfsPath, volumeID), mountpath, err)
+				return err
+			}
+		}
+	} else {
+		// Block access
+		if d.isRaw(v) {
+			return fmt.Errorf("Volume of raw format cannot be mounted")
+		}
+		if v.GetState() != api.VolumeState_VOLUME_STATE_ATTACHED {
+			return fmt.Errorf("Voume %s is not attached", volumeID)
+		}
+		mountExists, _ := d.mounter.Exists(v.DevicePath, mountpath)
+		if !mountExists {
+			if err := syscall.Mount(v.DevicePath, mountpath, v.Spec.Format.SimpleString(), 0, ""); err != nil {
```

Comment:
Mount should be no-op for raw block volumes.

Reply:
correct, it is returned at line 509.